kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableFeatures.java

```diff
@@ -136,10 +138,15 @@ public static void validateWriteSupportedTable(
         throw unsupportedWriterProtocol(tablePath, minWriterVersion);
       case 7:
         for (String writerFeature : protocol.getWriterFeatures()) {
+          // For version 7, we allow 'invariants' to be present in the protocol's writerFeatures
+          // to unblock certain use cases, provided that no invariants are defined in the schema.
+          if (writerFeature.equals(INVARIANTS_FEATURE_NAME)) continue;
           if (!SUPPORTED_WRITER_FEATURES.contains(writerFeature)) {
             throw unsupportedWriterFeature(tablePath, writerFeature);
           }
         }
+        validateNoInvariants(tableSchema);
```

Comment:
nit: personal preference
```suggestion
          if (writerFeature.equals(INVARIANTS_FEATURE_NAME)) {
            // For version 7, we allow 'invariants' to be present in the protocol's writerFeatures
            // to unblock certain use cases, provided that no invariants are defined in the schema.
            validateNoInvariants(tableSchema);
          } else if (!SUPPORTED_WRITER_FEATURES.contains(writerFeature)) {
            throw unsupportedWriterFeature(tablePath, writerFeature);
          }
        }
```